templates/daemonset.yaml

```diff
@@ -87,6 +87,13 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if and .Values.child.claiming.enabled .Values.child.claiming.token .Values.child.claiming.rooms }}
```

Comment:
as i said we need to add `.Values.child.persistUniqueID` to the `if and ...`

It makes it long and we use same condition in 3 places which makes the whole thing error-prone in addition.

Perhaps it would be better to create a variable and `if` it, like

```yml
{{ $addClaiming := and .Values.child.claiming.enabled .Values.child.claiming.token .Values.child.claiming.rooms .Values.child.persistUniqueID
...

{{- if $addClaiming }}
...

{{- if $addClaiming }}
...

{{- if $addClaiming }}
...

```

Just an idea. What do u think?

Reply:
> as i said we need to add `.Values.child.persistUniqueID` to the `if and ...`
> 
> It makes it long and we use same condition in 3 places which makes the whole thing error-prone in addition.
> 
> Perhaps it would be better to create a variable and `if` it, like
> 
> ```yaml
> {{ $addClaiming := and .Values.child.claiming.enabled .Values.child.claiming.token .Values.child.claiming.rooms .Values.child.persistUniqueID }}
> ...
> 
> {{- if $addClaiming }}
> ...
> 
> {{- if $addClaiming }}
> ...
> 
> {{- if $addClaiming }}
> ...
> ```
> 
> Just an idea. What do u think?

Love it. Count on it.